src/main/java/org/betonquest/betonquest/compatibility/citizens/EffectLibParticleManager.java

```diff
@@ -0,0 +1,166 @@
+                    continue;
+                }
+
+                final Set<Integer> npcs = loadNpcs(pack, settings);
+                final List<CompoundLocation> locations = loadLocations(pack, settings, key);
+                final List<ConditionID> conditions = loadConditions(pack, key, settings);
+
+                final EffectConfiguration effect = new EffectConfiguration(effectClass, locations, npcs, conditions, settings, conditionsCheckInterval);
+                final EffectLibRunnable particleRunnable = new EffectLibRunnable(effect);
+
+                activeParticles.add(particleRunnable);
+                particleRunnable.runTaskTimer(BetonQuest.getInstance(), 1, interval);
+            }
+        }
+    }
+
+    /**
+     * Reloads the particle effect.
+     */
+    public void reload() {
+        for (final EffectLibRunnable activeParticle : activeParticles) {
+            activeParticle.cancel();
+        }
+        activeParticles.clear();
+        loadParticleConfiguration();
+    }
+
+    private List<CompoundLocation> loadLocations(final QuestPackage pack, final ConfigurationSection settings, final String key) {
+        final List<CompoundLocation> locations = new ArrayList<>();
+        if (settings.isList("locations")) {
+            for (final String rawLocation : settings.getStringList("locations")) {
+                if (rawLocation == null) {
+                    continue;
+                }
+                try {
+                    locations.add(new CompoundLocation(pack, GlobalVariableResolver.resolve(pack, rawLocation)));
+                } catch (final InstructionParseException exception) {
+                    LOG.warn(pack, "Could not load npc effect '" + key + "' in package " + pack.getQuestPath() + ": "
+                            + "Location is invalid:" + exception.getMessage());
+                }
+            }
+        }
+        return locations;
+    }
+
+    @NotNull
+    private List<ConditionID> loadConditions(final QuestPackage pack, final String key, final ConfigurationSection settings) {
+        final List<ConditionID> conditions = new ArrayList<>();
+        for (final String conditionID : settings.getStringList("conditions")) {
+            try {
+                conditions.add(new ConditionID(pack, conditionID));
+            } catch (final ObjectNotFoundException e) {
+                LOG.warn(pack, "Error while loading npc_effects '" + key + "': " + e.getMessage(), e);
+            }
+        }
+        return conditions;
+    }
+
+    @NotNull
+    private Set<Integer> loadNpcs(final QuestPackage pack, final ConfigurationSection settings) {
+        final Set<Integer> npcs = new HashSet<>();
+        if (settings.isList(NPCS_CONFIG_SECTION)) {
+            npcs.addAll(settings.getIntegerList(NPCS_CONFIG_SECTION));
+        } else {
+            final String setting = settings.getString(NPCS_CONFIG_SECTION);
+            if (setting == null || !setting.equalsIgnoreCase("PACKAGE")) {
```

Comment:
I dont like the "PACKAGE" feature, because we propably remove the NPCs section from the packages in the future